pkg/etcdcli/etcdcli.go

```diff
@@ -294,7 +295,16 @@ func (g *etcdClientGetter) UnhealthyMembers() ([]*etcdserverpb.Member, error) {
 	for _, member := range membersResp.Members {
 		if len(member.ClientURLs) == 0 {
 			unhealthyMembers = append(unhealthyMembers, member)
-			unstartedMemberNames = append(unstartedMemberNames, member.Name)
+			unstartedName := member.Name
+			if unstartedName == "" {
+				u, err := url.Parse(member.PeerURLs[0])
+				if err == nil {
+					unstartedName = u.Hostname()
+				} else {
+					unstartedName = "NO-NAME-MEMBER"
```

Comment:
should this error be propagated up the stack? Is it possible for an admin to add a member and get caught here? Maybe more of a design decision that needs wider discussion, but okay for now